segmenters/image/FaceNetSegmenter/__init__.py

```diff
@@ -82,20 +81,30 @@ def segment(self, data: 'np.ndarray', *args, **kwargs) -> List[Dict]:
             # Create a batch of size 1
             image = image.unsqueeze(0)
 
-            faces, probabilities = self.face_detector(image, return_prob=True)
+            # Detect faces
+            batch_boxes, batch_probs, batch_points = self.face_detector.detect(image, landmarks=True)
+
+            # Select faces
+            if not self.keep_all:
+                batch_boxes, batch_probs, batch_points = self.face_detector.select_boxes(
+                    batch_boxes, batch_probs, batch_points, image, method=self.selection_method
+                )
 
-            if self.keep_all:
-                # All faces and probabilities are grouped in the first dimension of the first batch element
-                faces = faces[0]
-                probabilities = probabilities[0]
+            # Extract faces
+            faces = self.face_detector.extract(image, batch_boxes, save_path=None)
+            if faces[0] is not None:
+                faces = faces[0].view(-1, image.shape[-1], self.image_size, self.image_size)
+                batch_boxes = batch_boxes[0]
+                batch_probs = batch_probs[0]
 
             results = [
                 dict(
                     offset=0,
                     weight=probability,
-                    blob=face.detach().numpy(),
+                    blob=face.numpy(),
```

Comment:
are u sure about removing `detach`?

Reply:
yeah because using `with torch.no_grad()` context manager.

Reply:
btw followed the implementation from the `forward` function in their [library](https://github.com/timesler/facenet-pytorch/blob/master/models/mtcnn.py#L226) which uses bounding box and face extraction api internally .